test/plugins/wasm_bpf/wasm_bpf.cpp

```diff
@@ -0,0 +1,578 @@
+            WasmEdge::ValVariant(key), WasmEdge::ValVariant(value),
+            WasmEdge::ValVariant(nextKey), WasmEdge::ValVariant(flags)},
+        callResult));
+    return callResult[0].get<int32_t>();
+  };
+  // Three helper functions that will be used
+  auto mapGetNextKey = [&](int32_t fd, uint32_t lookupKey,
+                           uint32_t nextKey) -> int32_t {
+    // lookupKey is the last element -> returns -1
+    // lookupKey found -> returns 0, set nextKey
+    // lookupKey not found -> returns 0, set nextKey to the first key
+
+    return callMapOperate(fd,
+                          4, // BPF_MAP_GET_NEXT_KEY
+                          lookupKey, 0, nextKey, 0);
+  };
+  auto mapLookupElem = [&](int32_t fd, uint32_t key,
+                           uint32_t valueOut) -> int32_t {
+    // key found -> returns 0
+    // key not found -> returns -1
+    return callMapOperate(fd,
+                          1, // BPF_MAP_LOOKUP_ELEM
+                          key, valueOut, 0, 0);
+  };
+  auto mapDeleteElem = [&](int32_t fd, uint32_t key) -> int32_t {
+    // key found -> return 0
+    // key not found -> returns -1
+    return callMapOperate(fd,
+                          3, // BPF_MAP_DELETE_ELEM
+                          key, 0, 0, 0);
+  };
+  // Three helper functions to make read & write more convenient
+  auto readU32 = [&](uint32_t offset) -> uint32_t {
+    const auto *ptr = memoryInstRef.getPointer<const uint32_t *>(offset);
+    EXPECT_NE(ptr, nullptr);
+    return *ptr;
+  };
+  auto writeU32 = [&](uint32_t offset, uint32_t val) {
+    auto *ptr = memoryInstRef.getPointer<uint32_t *>(offset);
+    EXPECT_NE(ptr, nullptr);
+    *ptr = val;
+  };
+  auto readHistRef = [&](uint32_t offset) -> const hist & {
+    const auto *ptr = memoryInstRef.getPointer<const hist *>(offset);
+    EXPECT_NE(ptr, nullptr);
+    return *ptr;
+  };
+  const uint32_t lookUpKeyOffset = nextOffset;
+  nextOffset += sizeof(uint32_t);
+  const uint32_t nextKeyOffset = nextOffset;
+  nextOffset += sizeof(uint32_t);
+  const uint32_t histOffset = nextOffset;
+  nextOffset += sizeof(hist);
+
+  // Poll 10 times, with interval 1s
+  for (size_t i = 1; i <= 10; i++) {
+    using namespace std;
+    std::this_thread::sleep_for(std::chrono::seconds(1));
+    writeU32(lookUpKeyOffset, static_cast<uint32_t>(-2));
+    while (mapGetNextKey(histsFd, lookUpKeyOffset, nextKeyOffset) == 0) {
+      EXPECT_GE(mapLookupElem(histsFd, nextKeyOffset, histOffset), 0);
+      const auto &histRef = readHistRef(histOffset);
+      size_t maxIdx = 0;
+      for (size_t i = 0; i < std::size(histRef.slots); i++)
+        if (histRef.slots[i] > 0)
+          maxIdx = i;
+      for (size_t i = 0; i < maxIdx; i++) {
+        auto low = UINT64_C(1) << (i);
+        auto high = (UINT64_C(1) << (i + 1)) - 1;
+        cout.setf(ios::left);
+        cout << setw(6) << low << "..." << setw(6) << high << " " << setw(6)
```

Comment:
The same as above, there is no need to use the real data and output it. You can use asserts to check the maps, and use a simplify test case instead of the real eBPF example.

Reply:
For example: https://elixir.bootlin.com/linux/latest/source/tools/testing/selftests/bpf/test_maps.c

Write a simple eBPF program to create the maps and a simple program. Then test the map operations.